SConstruct

```diff
@@ -680,59 +821,13 @@ for variant_path in variant_paths:
         after_sconsopts_callbacks.append(cb)
     Export('AfterSConsopts')
 
-    # Sticky variables get saved in the variables file so they persist from
-    # one invocation to the next (unless overridden, in which case the new
-    # value becomes sticky).
-    sticky_vars = Variables(args=ARGUMENTS)
-    Export('sticky_vars')
-
-    # EXTRAS is special since it affects what SConsopts need to be read.
-    sticky_vars.Add(('EXTRAS', 'Add extra directories to the compilation', ''))
-
-    # Set env variables according to the build directory config.
-    sticky_vars.files = []
-    # Variables for $BUILD_ROOT/$VARIANT_DIR are stored in
-    # $BUILD_ROOT/$VARIANT_DIR/gem5.build/variables
-
-    gem5_build_vars = os.path.join(gem5_build, 'variables')
-    build_root_vars = os.path.join(build_root, 'variables', variant_dir)
-    current_vars_files = [gem5_build_vars, build_root_vars]
-    existing_vars_files = list(filter(isfile, current_vars_files))
-    if existing_vars_files:
-        sticky_vars.files.extend(existing_vars_files)
-        if not GetOption('silent'):
-            print('Using saved variables file(s) %s' %
-                    ', '.join(existing_vars_files))
-    else:
-        # Variant specific variables file doesn't exist.
-
-        # Get default build variables from source tree.  Variables are
-        # normally determined by name of $VARIANT_DIR, but can be
-        # overridden by '--default=' arg on command line.
-        default = GetOption('default')
-        opts_dir = Dir('#build_opts').abspath
-        if default:
-            default_vars_files = [
-                    gem5_build_vars,
-                    build_root_vars,
-                    os.path.join(opts_dir, default)
-                ]
-        else:
-            default_vars_files = [os.path.join(opts_dir, variant_dir)]
-        existing_default_files = list(filter(isfile, default_vars_files))
-        if existing_default_files:
-            default_vars_file = existing_default_files[0]
-            sticky_vars.files.append(default_vars_file)
-            print("Variables file(s) %s not found,\n  using defaults in %s" %
-                    (' or '.join(current_vars_files), default_vars_file))
-        else:
-            error("Cannot find variables file(s) %s or default file(s) %s" %
-                    (' or '.join(current_vars_files),
-                     ' or '.join(default_vars_files)))
-            Exit(1)
+    extras_file = os.path.join(gem5_build, 'extras')
+    extras_var = Variables(extras_file, args=ARGUMENTS)
+
+    extras_var.Add(('EXTRAS', 'Add extra directories to the compilation', ''))
```

Comment:
`EXTRAS` is missing from all of the new documentation. Is that on purpose?

I think we should at least mention how `EXTRAS` can be used now.

Reply:
Please see the commit message of this:
https://github.com/gem5/gem5/pull/69/commits/0ee2917562e3e234cda5b1921ae9c5d9656bd6f6